Add string type support to module.sources

dnohales opened the pull request:
According to flatpak-manifest manpage:

Additionally, the sources list can contain a plain string, which is interpreted
as the name of a separate json or yaml file that is read and inserted at
this point.

TingPing reviewed (changes requested):
That file is auto-generated so you need to modify `doc2json.py`. I realize that complicates this simple changes but otherwise it will be lost next time the file was regenerated.

dnohales commented:
Oops, sorry about that

dnohales committed:
Add string type support to module.sources

According to flatpak-manifest manpage:

Additionally, the sources list can contain a plain string, which is
interpreted as the name of a separate json or yaml file that is read and
inserted at this point.

dnohales commented:
Updated.
